pppwn.py

```diff
@@ -147,6 +147,30 @@ def handler(self, pkt):
             / PPPoE(sessionid=pkt[PPPoE].sessionid) / PPP() /
             PPP_LCP_Echo(code=ECHO_REPLY, id=pkt[PPP_LCP_Echo].id))
 
+def parse_pap(username, pwd, key):
+    if key == "sys_ver":
+        return pwd
+
+def getOffsets(sys_ver):
+    if sys_ver in ('750', '751', '755'):
+        return OffsetsFirmware_750_755()
+    elif sys_ver in ('800', '801', '803'):
+        return OffsetsFirmware_800_803()
+    elif sys_ver in ('850', '852'):
+        return OffsetsFirmware_850_852()
+    elif sys_ver == '900':
+        return OffsetsFirmware_900()
+    elif sys_ver in ('903', '904'):
+        return OffsetsFirmware_903_904()
+    elif sys_ver in ('950', '951', '960'):
+        return OffsetsFirmware_950_960()
+    elif sys_ver in ('1000', '1001'):
+        return OffsetsFirmware_1000_1001()
+    elif sys_ver in ('1050', '1070', '1071'):
+        return OffsetsFirmware_1050_1071()
+    elif sys_ver == '1100':
+        return OffsetsFirmware_1100()
+    return None
```

Comment:
```suggestion
def getOffsets(sys_ver):
    match sys_ver:
        case '750' | '751' | '755':
            return OffsetsFirmware_750_755()
        case '800' | '801' | '803':
            return OffsetsFirmware_800_803()
        case '850' | '852':
            return OffsetsFirmware_850_852()
        case '900':
            return OffsetsFirmware_900()
        case '903' | '904':
            return OffsetsFirmware_903_904()
        case '950' | '951' | '960':
            return OffsetsFirmware_950_960()
        case '1000' | '1001':
            return OffsetsFirmware_1000_1001()
        case '1050' | '1070' | '1071':
            return OffsetsFirmware_1050_1071()
        case '1100':
            return OffsetsFirmware_1100()
        case _:
            return None
```

Perhaps use a switch statement instead. Improves readability and also gives a slight perf improvement.

Reply:
Maybe use a map string -> function pointer